### PR TITLE
bug(checkbox): added missing readonly prop value

### PR DESCRIPTION
**Pull request by Rajdeepc**
## Description

Added missing readonly prop value

## Related issue(s)

- 

## Motivation and context

Showing an error on readonly prop in `CheckBox` swc-react component.

## How has this been tested?

`yarn link` at packages/checkbox in SWC

and `yarn link "@spectrum-web-components/checkbox"`

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Screenshots (if appropriate)

Screenshot 2023-12-08 at 2 22 05 PM

## Types of changes

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Review by Westbrook — COMMENTED** [review]
Is there a way to trigger the error that raised this? Being typings and not functionality, it seems like maybe not, but it would be good if we could add something that prevented issues like this in future refactors.

**Comment by Westbrook** [comment]
Related, looks like Spectrum CSS moves the `input` to `disabled` in this state, maybe we should do the same? 
image

File: packages/checkbox/src/CheckboxMixin.ts [hunk 1]
```diff
@@ -78,6 +83,19 @@ export function CheckboxMixin<T extends Constructor<ReactiveElement>>(
...
+        protected override updated(changes: PropertyValues): void {
+            super.updated(changes);
+            if (
+                changes.has('readonly') &&
+                (typeof changes.get('readonly') !== 'undefined' ||
+                    this.readonly)
+            ) {
+                if (this.readonly) {
+                    this.inputElement.disabled = this.readonly;
+                }
```
Comment (Westbrook) [inline packages/checkbox/src/CheckboxMixin.ts:96]: This doesn't look like it supports removing `readonly`?

Comment (Rajdeepc) [reply packages/checkbox/src/CheckboxMixin.ts]: Yup. great catch! update to allow support for both setting and removing readonly

File: packages/checkbox/src/CheckboxMixin.ts [hunk 2]
```diff
@@ -78,6 +83,17 @@ export function CheckboxMixin<T extends Constructor<ReactiveElement>>(
...
         }
+
+        protected override updated(changes: PropertyValues): void {
+            super.updated(changes);
+            if (
+                changes.has('readonly') &&
+                (typeof changes.get('readonly') !== 'undefined' ||
+                    this.readonly)
+            ) {
+                this.inputElement.disabled = this.readonly;
```
[the remark this answers is not in the document]
Comment (Rajdeepc) [reply packages/checkbox/src/CheckboxMixin.ts:94]: ok added the disabled property in the render:
```
protected render(): TemplateResult {
            return html`
                <input
                    id="input"
                    name=${ifDefined(this.name || undefined)}
                    type="checkbox"
                    .checked=${this.checked}
                    ?disabled=${this.readonly} 
                    @change=${this.handleChange}
                />
            `;
        }
```

**Review by Westbrook — COMMENTED** [review]
Is there any way that we can test the specific issue that arose here in your consumption?

**Comment by Rajdeepc** [comment]
> Is there any way that we can test the specific issue that arose here in your consumption?

Yes should be added now.

**Pull request merged**